setup.py

```diff
@@ -1,106 +0,0 @@
-                line = line.strip()
-                if not line or line.startswith("#"):
-                    continue
-                requirements.append(line)
-    return requirements
-
-
-def extract_version():
-    version = None
-    fname = os.path.join(PACKAGE_DIR, 'iris_grib', '__init__.py')
-    with open(fname) as fi:
-        for line in fi:
-            if (line.startswith('__version__')):
-                _, version = line.split('=')
-                version = version.strip()[1:-1]  # Remove quotations
-                break
-    return version
-
-
-def long_description():
-    fname = os.path.join(PACKAGE_DIR, "README.rst")
-    with open(fname, "rb") as fi:
-        result = fi.read().decode("utf-8")
-    return result
-
-
-def file_walk_relative(top, remove=''):
-    """
-    Returns a generator of files from the top of the tree, removing
-    the given prefix from the root/file result.
-
-    """
-    top = top.replace('/', os.path.sep)
-    remove = remove.replace('/', os.path.sep)
-    for root, dirs, files in os.walk(top):
-        for file in files:
-            yield os.path.join(root, file).replace(remove, '')
-
-
-setup_args = dict(
-    name             = PYPI_NAME,                                # noqa: E221,E251,E501
-    version          = extract_version(),                        # noqa: E221,E251,E501
-    packages         = packages,                                 # noqa: E221,E251,E501
-    package_data     = {'iris_grib': list(                       # noqa: E221,E251,E501
-        file_walk_relative('iris_grib/tests/results', remove='iris_grib/'))
-    },
-    description      = "GRIB loading for Iris",                  # noqa: E221,E251,E501
-    long_description = long_description(),                       # noqa: E221,E251,E501
-    long_description_content_type = "text/x-rst",                # noqa: E221,E251,E501
-    url              = 'https://github.com/SciTools/iris-grib',  # noqa: E221,E251,E501
-    author           = 'UK Met Office',                          # noqa: E221,E251,E501
-    author_email     = 'scitools-iris@googlegroups.com',         # noqa: E221,E251,E501
-    license          = 'BSD',                                    # noqa: E221,E251,E501
-    platforms        = "Linux, Mac OS X, Windows",               # noqa: E221,E251,E501
-    keywords         = ['iris', 'GRIB'],                         # noqa: E221,E251,E501
-    classifiers=[                                                # noqa: E221,E251,E501
-        'Programming Language :: Python',
-        'Programming Language :: Python :: 3.9',
-        'Programming Language :: Python :: 3.10',
-        'Programming Language :: Python :: 3.11',
-        'Programming Language :: Python :: 3 :: Only',
-    ],
-    # NOTE: The Python 3 bindings to eccodes (eccodes-python) is available on
-    # PyPI, but the user is required to install eccodes itself manually. See
-    # ECMWF ecCodes installation documentation for more information.
```

Comment:
Looks like this is no longer true 👍
(I just tested installing using your `pyproject.toml` and it successfully sorted out `eccodes`).